homeassistant/components/insteon/insteon_entity.py

```diff
@@ -0,0 +1,127 @@
+"""Insteon base entity."""
+import logging
+
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    DOMAIN,
+    INSTEON_ENTITIES,
+    SIGNAL_LOAD_ALDB,
+    SIGNAL_PRINT_ALDB,
+    STATE_NAME_LABEL_MAP,
+)
+from .utils import print_aldb_to_log
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class InsteonEntity(Entity):
+    """INSTEON abstract base entity."""
+
+    def __init__(self, device, state_key):
+        """Initialize the INSTEON binary sensor."""
+        self._insteon_device_state = device.states[state_key]
+        self._insteon_device = device
+        self._insteon_device.aldb.add_loaded_callback(self._aldb_loaded)
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def address(self):
+        """Return the address of the node."""
+        return self._insteon_device.address.human
+
+    @property
+    def group(self):
+        """Return the INSTEON group that the entity responds to."""
+        return self._insteon_device_state.group
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        if self._insteon_device_state.group == 0x01:
+            uid = self._insteon_device.id
+        else:
+            uid = "{:s}_{:d}".format(
+                self._insteon_device.id, self._insteon_device_state.group
+            )
+        return uid
+
+    @property
+    def name(self):
+        """Return the name of the node (used for Entity_ID)."""
+        # Set a base description
+        description = self._insteon_device.description
+        if self._insteon_device.description is None:
+            description = "Unknown Device"
+
+        # Get an extension label if there is one
+        extension = self._get_label()
+        if extension:
+            extension = f" {extension}"
+        name = "{:s} {:s}{:s}".format(
+            description, self._insteon_device.address.human, extension
+        )
+        return name
+
+    @property
+    def device_state_attributes(self):
+        """Provide attributes for display on device card."""
+        attributes = {"INSTEON Address": self.address, "INSTEON Group": self.group}
```

Comment:
State attribute keys should be lowercase snakecase.

Reply:
I made this change but don't know how to add the "breaking change" flag.

Reply:
I've added the label. Please write a breaking change paragraph in the PR description. Briefly describe what has changed and what the user needs to do to cope with the breaking change.